src/cp.js

```diff
@@ -153,6 +171,42 @@ function cpdirSyncRecursive(sourceDir, destDir, currentDepth, opts) {
   } // for files
 } // cpdirSyncRecursive
 
+function major(rdev) {
+  return ((rdev >> 31 >> 1) & 0xfffff000) | ((rdev >> 8) & 0x00000fff);
+}
+function minor(rdev) {
+  return ((rdev >> 12) & 0xffffff00) | (rdev & 0x000000ff);
+}
+
+// attempts to create a dev (character or block) by shelling out to mknod.
+function mknod(destFile, stat) {
+  var created;
+  var failureMessage;
+  var args = ['mknod'];
+  args.push('"' + destFile + '"');
+  if (stat.isFIFO()) {
+    args.push('p');
+  } else {
+    if (stat.isCharacterDevice()) args.push('c');
+    if (stat.isBlockDevice()) args.push('b');
+    args.push(major(stat.rdev), minor(stat.rdev));
+  }
+  try {
+    var cmd = exec(args.join(' '), { silent: true });
```

Comment:
For security, I recommend `child_process.execFileSync()` here. Just wrap it in a try-catch, since it throws an exception if the command fails.

```js
try {
  child_process.execFileSync('mknod', argsArray);
} catch (e) {
  failureMessage = e.stderr;
}
```